Gems/Atom/RHI/Code/Source/RHI.Edit/ShaderCompilerArguments.cpp

```diff
@@ -155,11 +155,6 @@ namespace AZ
             {
                 arguments += " -Zpr";
             }
-            if (m_dxcGenerateDebugInfo)
```

Comment:
Note for the reviewer:

I opted to move this out of this function because `BuildHasDebugInfo` is technically virtual and overridden per shader platform interface. Checking `m_dxcGenerateDebugInfo` directly and adding debug arguments here will bypass any behavior used to override/bypass the debug-info detection.